### DISCUSSION TITLE
Request: Will there be some community RPC services support Darwinia2 & Crab2?

### DISCUSSION THREAD

**Discussion by 1022ranji** [ringdao-community-guild/discussion#11] [General]
It would be better if there will be some community RPC services support for Darwinia2 & Crab2.

I think the benefits of community RPC services are as follows:

Increased security: It can prevent malicious attacks and the possibility of the network being attacked by verifying and processing transactions. The distributed structure of RPC services can prevent single-point failures and improve the overall security of the network.

Improved efficiency: Community RPC services can improve the efficiency of the entire blockchain network by processing transactions and storing block data. Nodes can quickly confirm transactions and broadcast them to other RPC services, improving transaction processing speed.

Increased decentralization: Community RPC services can enable more individuals or organizations to participate in the blockchain network, making the entire network more decentralized. This will help protect users' rights and improve the stability of the blockchain network.

Promoting community development: Community RPC services can promote communication and collaboration among community members, thereby promoting the development and growth of the entire blockchain community. 

In summary, community RPC services are very helpful for the security, efficiency, decentralization, and community development of blockchain networks.

**Comment by furoxr** [comment]
Good idea, maybe I could run a collator node, Is there any tutorial I could follow?

**Comment by hackfisher** [comment]
https://github.com/darwinia-network/community-rpc

@fewensa maintained this repo, Its solution was designed to fit in this initiative, maybe we can move the repo here for following up maintainance.

**Comment by fewensa** [comment]
**Proposal: Community RPC Experimental Program**
===


## What is Community RPC?

The core purpose of the Community RPC Experimental Program is to fully utilize community resources and provide a stable and reliable RPC service supplement.

When a member becomes a Darwinia or Crab Collator, or simply runs a node, they need to prepare a computer to support it. If they only run it as a node, there will be a lot of idle resources. The goal of the Community RPC program is to make use of these idle resources and provide rewards based on the quality of RPC operation. This is to incentivize community members to participate and provide better service quality, creating a positive cycle to deliver a highly available and stable RPC service.

To achieve this, we need to implement:

- Provide an entry for community members to submit their RPC addresses
  - [community-rpc](https://github.com/dcdao/community-rpc)
- Provide an RPC proxy solution to facilitate the proxying of Community RPC
- Analyze the reward data for the submitted RPC addresses by community members
- Distribute rewards to the members participating in Community RPC based on the data analysis results


## Tasks

To accomplish this, we will divide the work into the following steps.

### Step 1 - RPC Endpoint Submission Entry

Set up an entry for members to submit RPC endpoints. Members should follow the guidelines for endpoint submission. [Submit RPC endpoint](https://github.com/dcdao/community-rpc/issues/new/choose)
After a community member submits an endpoint, the DCDAO members will verify it and add the endpoint to the proxy list.

### Step 2 - Automated Deployment and Updates

Whenever a new endpoint is added or there are any changes to the proxy list, the backend proxy list should be automatically updated. The proxy service should also shield inactive proxy nodes to ensure the quality of the RPC service.

### Step 3 - Service Quality Validation

All planned endpoints that have joined should undergo service quality validation, including regular access checks, etc.
Furthermore, the validation of service quality is also the basis for subsequent reward distribution. More refined rules should be determined based on actual proxy data statistics.


## Requirements

1. DCDAO members need to monitor [community-rpc/issues](https://github.com/dcdao/community-rpc/issues) for any new members joining the program and respond accordingly.
2. DevOps personnel should monitor the CI status of the community-rpc repository and pay attention to the service quality of the endpoints, taking appropriate actions.
3. Provide statistics on the service quality of endpoints as a basis for reward distribution.


## Budget

The Community RPC Experimental Program will apply for 3000U worth of RING as rewards per quarter. The rewards will be distributed based on the performance statistics of participating members at the end of each quarter.

Address: `0x3198D3070D84cB6798651AA3c02828Ba5866b296`
Chain: Arbitrum

**Comment by hackfisher** [comment]
LGTM.

Suggest to do reward distribution chain on Arbitrum?

**Comment by fewensa** [comment]
https://snapshot.org/#/darwiniacommunitydao.eth/proposal/0x26be609917096095e8fb2a617d6def3fa62b3240285a9887bcd9184afb2a0ddd

**Comment by hackfisher** [comment]
wss://crab-community-rpc.darwinia.network

https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fcrab-community-rpc.darwinia.network#/explorer
